lib/slack-rtmapi.rb

```diff
@@ -17,4 +23,39 @@ def self.get_url(options)
     URI(body['url'])
   end
 
+  def initialize(options)
+    @token = options.fetch(:token)
+    @api_url = options[:slack_api_url] || 'https://slack.com/api'
+  end
+
+  def start_rtm
+    @data = SlackWebAPI.call(method: 'rtm.start', token: @token)
+    @websocket_url = URI(@data['url'])
+    client
+  end
+
+  def client
+    @client ||= SlackRTM::Client.new websocket_url: @websocket_url
+  end
+
+  def channel_id(channel_name)
+    channels = @data['channels']
+    channel_data = channels.find { |channel| channel['name'] == channel_name }
+    channel_data['id'] unless channel_data.nil?
+  end
+end
+
+#
+# SlackWebAPI
+#
+module SlackWebAPI
+  def self.call(options)
+    options.fetch(:token)
+    method = options.fetch(:method)
+    fail unless method.match(/^[a-z]+\.[A-z]+$/)
```

Comment:
I think that slack has some underscores in their method names.
Maybe `/^[\w_]+\.[\w_]+$/` would be enough ? (even if it's wider than the real range of acceptable characters)